lib/puppet/defaults.rb

```diff
@@ -1280,7 +1280,7 @@ def self.initialize_default_settings!(settings)
     }
   )
 
-  settings.define_settings(:master,
+  settings.define_settings(:server,
```

Comment:
One side-effect of defining settings in a new section is that calls to `Puppet.settings.use(:master)` will no longer apply the settings catalog for file/directory settings in the `[master]` section. It doesn't appear puppetserver, puppetdb terminus or classifier terminus ever call that method. But to be ensure compatibility in 6.x, we should modify `Puppet.settings.use` so it always applies both the `:master` and `:server` sections if either are specified. Also thinking we should issue a deprecation warning if `Puppet.settings.use` is called with `:master`.